MiniDB/DBTransaction.cs

```diff
@@ -0,0 +1,145 @@
+        /// </summary>
+        private static int nextAvailableID = 0;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DBTransaction{T}" /> class.
+        /// </summary>
+        public DBTransaction() : base()
+        {
+            this.ID = nextAvailableID;
+            this.Transaction_timestamp = System.DateTime.Now;
+            this.Active = true;
+            nextAvailableID++;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DBTransaction{T}" /> class.
+        ///   Copy constructor by value
+        /// </summary>
+        /// <param name="inTransaction">The transaction to copy</param>
+        internal DBTransaction(DBTransaction<T> inTransaction) : base()
+        {
+            this.ID = inTransaction.ID;
+            this.Transaction_timestamp = inTransaction.Transaction_timestamp;
+            this.Transacted_item = inTransaction.Transacted_item;
+            this.TransactionType = inTransaction.TransactionType;
+            this.Item_ID = inTransaction.Item_ID;
+            this.Changed_property = inTransaction.Changed_property;
+            this.Property_new = inTransaction.Property_new;
+            this.Property_old = inTransaction.Property_old;
+            
+            // needed? - keep until collision detecting is implemented
+            nextAvailableID++;
+        }
+
+        #region transacted object properties
+        /// <summary>
+        /// Gets or sets the ID of the item that was acted on
+        /// </summary>
+        public ID Item_ID { get; set; }
+
+        /// <summary>
+        /// Gets or sets the object that was acted on
+        /// </summary>
+        public T Transacted_item { get; set; }
+
+        /// <summary>
+        /// Gets or sets what property was changed
+        /// </summary>
+        public string Changed_property { get; set; }
+
+        /// <summary>
+        /// Gets or sets the old property value
+        /// </summary>
+        public object Property_old { get; set; }
+
+        /// <summary>
+        /// Gets or sets the new property value
+        /// </summary>
+        public object Property_new { get; set; }
+        #endregion
+
+        #region transaction properties
+        /// <summary>
+        /// Gets an ID of this transaction (hides default ID, as we want this to be incremental).
+        /// </summary>
+        [JsonProperty]
+        public int ID { get; private set; } // switch to using default ID??
+
+        /// <summary>
+        /// Gets the system timestamp that this transaction occured at.
+        /// </summary>
+        [JsonProperty]
+        public System.DateTime Transaction_timestamp { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the type of transaction (Add, delete, modify, etc.) of type <see cref="TransactionType"/>.
+        /// </summary>
+        public TransactionType TransactionType { get => this.Get(); set => this.Set(value); }
```

Comment:
You probably want some kind of comment explaining why some of these properties are using the Get() and Set() while others don't need to.